packages/vscode-extension/lib/preview.js

```diff
@@ -44,25 +45,66 @@ function getComponentName({ type }) {
   return "(unnamed)";
 }
 
+async function getSourceFromComponent(component) {
+  if (component._source) {
+    return component._source;
+  }
+
+  const debugStack = component._debugStack;
+  const parsedStack = RNInternals.parseErrorStack(debugStack.stack);
+
+  const { file } = parsedStack[1];
+
+  const url = new URL(file);
+  const metroAddress = url.origin;
+
+  const metroSymbolicateResponse = await fetch(
```

Comment:
maybe we can use `react-native/Libraries/Core/Devtools/symbolicateStackTrace` via RNInternals here?

Reply:
we'd need to use `parseErrorStack` for the above to work, but should be ok